sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -883,6 +883,47 @@ createKernelArgMask(const pi::ByteArray &Bytes) {
   return Result;
 }
 
+/// This will setup environment variable for XRT to match the type of kernels
+/// being loaded.
+static void setupEnvironmentForKernels(RTDeviceBinaryImage *Img) {
+  static bool has_been_invoked = false;
+  const char* target = Img->getTarget();
+  constexpr const char *env_var = "XCL_EMULATION_MODE";
+  if (strncmp("fpga64_", target, 7) == 0 ||
+      strncmp("fpga32_", target, 7) == 0) {
+    target += 7;
+    if (strcmp(target, "hw") == 0) {
+      if (has_been_invoked)
+        if (const char *mode = std::getenv(env_var))
```

Comment:
no because all throws can happens only when has_been_invoked and during the first pass through the variable will be changed to match what the first kernel needs.